sklearn/decomposition/dict_learning.py

```diff
@@ -741,26 +716,46 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
                                   alpha=alpha, n_jobs=n_jobs).T
 
         # Update the auxiliary variables
-        if ii < batch_size - 1:
-            theta = float((ii + 1) * batch_size)
+        if online:
+            if ii < batch_size - 1:
+                theta = float((ii + 1) * batch_size)
+            else:
+                theta = float(batch_size ** 2 + ii + 1 - batch_size)
+            beta = (theta + 1 - batch_size) / (theta + 1)
         else:
-            theta = float(batch_size ** 2 + ii + 1 - batch_size)
-        beta = (theta + 1 - batch_size) / (theta + 1)
-
+            beta = 0.
         A *= beta
         A += np.dot(this_code, this_code.T)
         B *= beta
         B += np.dot(this_X.T, this_code.T)
```

Comment:
I think it's better to move the moving average update inside the `if online` case and do 
```
else:
    A = np.dot(...)
    B = np.dot(...)
```

just because it's less subtle